tensorflow/python/kernel_tests/cwise_ops_test.py

```diff
@@ -587,7 +612,8 @@ def _compareBoth(self, x, y, np_func, tf_func, also_compare_variables=False):
     self._compareCpu(x, y, np_func, tf_func, also_compare_variables)
     if x.dtype in (np.float16, np.float32, np.float64):
       if tf_func not in (_FLOORDIV, math_ops.floordiv, math_ops.igamma,
-                         math_ops.igammac, math_ops.zeta, math_ops.polygamma):
+                         math_ops.igammac, math_ops.zeta, math_ops.polygamma,
+                         _MATMUL):
```

Comment:
You shouldn't need to test gradients for `@`, since it's just a wrapper around `matmul`.  Gradient tests are for new ops.  This complexity will go away once you make a new file with targeted tests of `@`.